### PR TITLE
feat(v2): rollout announcement

### PR DESCRIPTION
**Pull request by hanstirtaputra**
## Problem

This PR adds a rollout announcement modal to the home screen to show users the new features that are available 

Closes [ #3906 ]

## Solution

**Breaking Changes** 
- [ ] Yes - this PR contains breaking changes
- [x] No - this PR is backwards compatible  

**Features**:

- feat: add rollout announcement modal

**New dependencies**:

- [lottie-web](https://www.npmjs.com/package/lottie-web) for Lottie animations

## Before & After Screenshots

New story added as `RolloutAnnouncement` in `Pages`

### CODE REVIEW

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (mantariksh) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:41]: can destructure this in the function arguments
```suggestion
  { numIndicators, currActiveIdx, onClick }: ProgressIndicatorProps,
): JSX.Element => {
```

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (mantariksh) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:35]: why make this optional?

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (mantariksh) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:56]: nit: imo slightly more readable if inlined
```suggestion
          return (
            <CircleIndicator
              isActiveIndicator={idx === currActiveIdx}
```

**Comment by karrui** [comment]
can there also be a storybook story for visual snapshot testing? see modal stories for examples of how to render just the modal

File: package.json
Comment (hanstirtaputra) [reply package.json]: If we use `framer-motion`, I think we need to re-create the animations from the `.svg`s. It won't be as simple as using the `.json`s provided and plugging it into the `lottie-web` library. I think it's possible, do we want to try that instead?

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx]: Okay updated the PR to do this, thanks!

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx]: Initially thought there could be a scenario where the progress indicators don't need to be clickable. But now I think it wouldn't be consistent design-wise. I've updated the PR so that `onClick` is no longer optional

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx]: Updated the PR as per your suggestion, thanks!

File: package.json
Comment (karrui) [reply package.json]: I think it's fine, just use Lottie it's the standard for animations sadly

File: frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx
Comment (mantariksh) [inline frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx:32]: let's be consistent about casing for constants, seems like `numOtherUpdatesPages` should also be `UPPER_SNAKE_CASE`. also we can define these outside the component.

File: frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx
Comment (mantariksh) [inline frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx:33]: some food for thought: @karrui uses `useMemo` for everything for reasons listed in [this article](https://attardi.org/why-we-memo-all-the-things/), but [some would disagree](https://kentcdodds.com/blog/usememo-and-usecallback). will leave it up to you! (or @karrui if he reviews)

File: frontend/src/features/rollout-announcement/components/LastFeatureContent.tsx
Comment (mantariksh) [inline frontend/src/features/rollout-announcement/components/LastFeatureContent.tsx:15]: can destructure in the function arguments directly

File: frontend/src/constants/localStorage.ts
Comment (mantariksh) [inline frontend/src/constants/localStorage.ts:19]: hmmm...what if two users use the same browser? only the first one will see the announcement right?

Comment (hanstirtaputra) [reply frontend/src/constants/localStorage.ts]: Yes only the first one will see. I've updated the key to be suffixed with user's ID, this change solves this problem

File: frontend/src/features/rollout-announcement/components/LastFeatureContent.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/LastFeatureContent.tsx]: Okay, updated PR to destructure in the function args

File: frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx:33]: Decided not to `useMemo` for this because I think it's not an expensive calculation. But if we want to be consistent as suggested in the [article](https://attardi.org/why-we-memo-all-the-things/), I'll update the PR again

File: frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx]: Updated the PR to fix this

File: frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx:73]: nit: should use rem instead of number for size. 24 === 1.5rem

File: frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx:64]: nit: use stack instead of box

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:30]: for accessibility, we should cast this box into a button:

```tsx
<Box
  // other props
  as="button"
/>
```

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:63]: nit: can use `MotionBox` component, which then allows usage of chakra props:

```tsx
<MotionBox
  pos='absolute'
  animate={animationProps}
  transition={{ stiffness: 100 }}
>
  ...
</MotionBox>
```

prefer that over inline styles

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:55]: should we extract out this into a memoized constant

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:50]: nit: useMemo everything pls

File: frontend/src/features/rollout-announcement/index.ts
Comment (karrui) [inline frontend/src/features/rollout-announcement/index.ts:1]: prefer non-default exports, read https://blog.neufund.org/why-we-have-banned-default-exports-and-you-should-do-the-same-d51fdc2cf2ad?gi=4bb93b11b

the few default exports we have in the codebase is due to dynamic imports for chunking, which this component does not need

File: frontend/src/features/workspace/WorkspacePage.tsx
Comment (karrui) [inline frontend/src/features/workspace/WorkspacePage.tsx:113]: usememo, maybe also don't set a key if there is no id yet, so nothing gets saved in localstorage as `xxxxxxxxundefined`

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:13]: Top padding wouldn't have the correct `primary.100` background colour though

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:62]: The `ModalHeader` has default paddings for start and end though, if we inline the `LottieAnimation` then we'll need to change the `ModalHeader` paddings, which we want to have for the title

File: frontend/src/features/rollout-announcement/components/LastFeatureContent.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/LastFeatureContent.tsx]: Okay thanks

File: frontend/src/features/rollout-announcement/components/LastFeatureContent.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/LastFeatureContent.tsx]: I was using the values in Figma and divide by 16 to get the `rem` values. I'll round to nearest 0.25 then

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx]: Okay thanks, updated PR to have `LottieAnimationProps` extend `BoxProps`

File: frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx]: I was trying to remove the appearance of hover effects, I realised that we should use `badge` instead since, updated the PR to use `badge`

File: frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx]: Okay thanks, updated to use `Button`

File: frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx]: I updated to use `Flex` since it's in a row for desktop view

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx]: Okay, updated to use `MotionBox` instead of `motion.div`

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx]: Yep, I updated to extract into a memoized constant

File: frontend/src/features/rollout-announcement/index.ts
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/index.ts]: Ahh okay thanks, I updated to use non-default export

File: frontend/src/features/workspace/WorkspacePage.tsx
Comment (hanstirtaputra) [reply frontend/src/features/workspace/WorkspacePage.tsx]: Okay, I updated to only show announcement if `useUser` loading is done. So the key can only be set once there's a userId since key is set when announcement is closed

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (karrui) [reply frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:13]: gotcha. keep it then,

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (karrui) [reply frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:63]: we can update padding instead. should strive for cleaner html

File: frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx
Comment (karrui) [reply frontend/src/features/rollout-announcement/RolloutAnnouncementModal.tsx]: Stack has two directions (you can set `direction='row'` for row flex direction, and also provides padding. In fact, Stack is a flexbox!

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:61]: nit: can use `py` over explicit paddingTop/Bottom props

File: frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx:11]: TIL `columnGap` (and `gap`) works on flex

File: frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx:1]: we only use boxicons. suspect this is the wrong icon to use.

File: frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx:15]: nit

```suggestion
      <Text color="secondary.500" textStyle="caption-1">
```

File: frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx:9]: nit: we can just put color="secondary.500" on the parent Badge component and do not need extra declarations in the children components.

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:49]: we don't need this constant; should just inline empty string in the `indicator` variable?

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:55]: can also inline this constant

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:62]: these two variables do not need to be in separate useMemos haha. they can be in a single one

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:64]: I actually added the padding to match the Figma though 🤔 , did feel a little weird. But I'll remove it

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:49]: Yep sounds good, was thinking not to have magic numbers when I wrote this, but guess this one is pretty clear

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:62]: Right... okay thanks

File: frontend/src/features/workspace/WorkspacePage.tsx
Comment (hanstirtaputra) [reply frontend/src/features/workspace/WorkspacePage.tsx]: Nice, thanks

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:69]: Yep it does, updated the PR to add explanation comment

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:13]: looking at it, i still don't think this is needed. the appropriate padding and bg can be put in the `LottieAnimation` component

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:40]:
```suggestion
  return <Box bg="primary.100" pt="4.5rem" {...boxProps} ref={element} />
```

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:60]: nit: use margins instead for spacing `mt`. The value required seems to be 0.625rem (10px) instead of 8.

File: frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureTag.tsx:16]: nit: don't think textStyle is required here. the base component has the text style already included.
nit: don't think you even need to wrap the text in a `Text` component.

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:32]: add aria-label for accessibility e.g. aria-label="Page X of Y". might want to ask @pearlyong for focus styles for the pagination dots.

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:11]: might want to make this a button too (that does nothing), and add aria-label and focus styles too. these can maybe be in a later PR if designers don't mind

**Review by karrui — APPROVED** [review]
lgtm, wait for designers to approve

File: frontend/src/features/rollout-announcement/assets/LogicOrderSvgr.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/assets/LogicOrderSvgr.tsx:44]: this will output errors, please fix before merging. you need to run the original svg through https://react-svgr.com/playground/, not just wrap it in a react component haha. that's why its called svgr. same for the other svgs.

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:13]: Right, removed TopSpacer and used padding and bg instead

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:60]: Okay updated back to `0.625rem`, I rounded off to nearest 0.25rem

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:32]: Okay I'll add aria-label and focus styles in a separate PR. I created an [issue](https://github.com/opengovsg/FormSG/issues/3977) to track this

File: frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/components/ProgressIndicator.tsx:11]: Created an [issue](https://github.com/opengovsg/FormSG/issues/3977) to track this, will add in a separate PR

File: frontend/src/features/rollout-announcement/assets/LogicOrderSvgr.tsx
Comment (hanstirtaputra) [reply frontend/src/features/rollout-announcement/assets/LogicOrderSvgr.tsx]: I see.. okay updated

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:16]: does this need an initializer? can leave undefined imho

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx:11]: should we extend this (and the `animationData` prop too) from lottie types?

File: frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx
Comment (karrui) [inline frontend/src/features/rollout-announcement/components/NewFeatureContent.tsx]: nit: prefer using chakra's responsive styles syntax whenever possible (using isMobiles requires some computation of current width instead of using media queries)

```tsx
height={['33%']}
// not sure it the above will work properly, if not can try
// height={{ base: 33%, md: 'initial' }}
```

**Pull request merged**
